src/thor/alternates.cc

```diff
@@ -0,0 +1,128 @@
+#include <iostream>
+#include <vector>
+
+#include "thor/alternates.h"
+
+using namespace valhalla::thor;
+using namespace valhalla::baldr;
+using namespace valhalla::midgard;
+
+/*
+ * Viability tests for alternate paths based on M. Kobitzsch's Alternative Route
+ * Techniques (2015). Tests verify limited sharing between segments, bounded
+ * stretch, and local optimiality. Any candidate path that meets all the criteria
+ * may be considered a valid alternate to the shortest path.
+ */
+namespace {
+// Defaults thresholds
+float kAtMostLonger = 1.25f; // stretch threshold
+float kAtMostShared = 0.75f; // sharing threshold
```

Comment:
Noting that we chose these thresholds based on OSRM. It may be worth tweaking them in the future, or even exposing a request option to customize them: https://github.com/Project-OSRM/osrm-backend/blob/master/src/engine/routing_algorithms/alternative_path_mld.cpp#L50